tracer/src/Datadog.Trace/Generated/net461/Datadog.Trace.SourceGenerators/Datadog.Trace.SourceGenerators.TelemetryMetric.TelemetryMetricGenerator/MetricsTelemetryCollector_count.g.cs

```diff
@@ -616,6 +632,16 @@ private static MetricKey[] GetCountBuffer()
             new(new[] { "type:timeout" }),
             new(new[] { "type:network" }),
             new(new[] { "type:status_code" }),
+            // waf.init, index = 413
+            new(null),
+            // waf.updates, index = 414
+            new(null),
+            // waf.requests, index = 415
+            new(new[] { "waf_version", "rule_triggered:false", "request_blocked:false", "waf_timeout:false", "request_excluded:false" }),
+            new(new[] { "waf_version", "rule_triggered:true", "request_blocked:false", "waf_timeout:false", "request_excluded:false" }),
+            new(new[] { "waf_version", "rule_triggered:true", "request_blocked:true", "waf_timeout:false", "request_excluded:false" }),
+            new(new[] { "waf_version", "rule_triggered:false", "request_blocked:false", "waf_timeout:true", "request_excluded:false" }),
+            new(new[] { "waf_version", "rule_triggered:false", "request_blocked:false", "waf_timeout:false", "request_excluded:true" }),
```

Comment:
Also I assume the other combinations of booleans aren't possible, right?

Reply:
> but can't we hardcode the WAF version

I assumed we would already have done that if we could, but this is suspicious:

> As [we hardcode it in our build pipeline](https://github.com/DataDog/dd-trace-dotnet/pull/4112/files)

I guess we can't guarantee the libddwaf version in version conflict scenarios, so _probably_ need to do this at runtime unfortunately. But I agree, it would simplify a lot of things if not

> Also I assume the other combinations of booleans aren't possible, right?

According to Robert 🙂 If he finds more options when implementing the metrics then we can always add them!

Reply:
> I guess we can't guarantee the libddwaf version in version conflict scenarios, so probably need to do this at runtime unfortunately.

😱 the only place where we use the waf is in auto instru, right? So version conflict shouldn't kick in in that case, right? 🙏

Reply:
> the only place where we use the waf is in auto instru, right

Well, _technically_ not in version conflict right? Security will still be initialized in that case, but I agree, we _shouldn't_ get any metrics 🤔 

I guess this is a solution for now, and if we can revisit and make that a constant, we can simplify later? 🙂

Reply:
Added here: https://github.com/DataDog/dd-trace-dotnet/pull/4293/commits/42671f2f45d9c05728bd0fabef72e1ca44aa9263

Reply:
> I guess this is a solution for now, and if we can revisit and make that a constant, we can simplify later?

Obviously. I'm not challenging to change all that in that PR. But I think I need to discuss with @robertpi :)